File: mc/rmcp/src/main/java/calebxzhou/rdi/mc/common2/mcp/RQuestChapterList.java
package calebxzhou.rdi.mc.common2.mcp;

import java.util.List;

public record RQuestChapterList(
        int total,
        int visible,
        List<Chapter> chapters
) {
    public record Chapter(
            String id,
            String title,
            String groupId,
            String groupTitle,
            boolean visible,
            boolean started,
            boolean completed,
            int progress,
            int questCount,
            int visibleQuestCount,
            int startableQuestCount,
            int completedQuestCount
    ) {
    }
}

File: mc/rmcp/common/src/main/kotlin/calebxzhou/rdi/mc/common2/mcp/RQuestChapterList.kt
package calebxzhou.rdi.mc.common2.mcp

@JvmRecord
data class RQuestChapterList(
    val total: Int = 0,
    val visible: Int = 0,
    val chapters: List<Chapter> = listOf()
) {
    @JvmRecord
    data class Chapter(
        val id: String,
        val title: String,
        val groupId: String,
        val groupTitle: String,
        val visible: Boolean,
        val started: Boolean,
        val completed: Boolean,
        val progress: Int,
        val questCount: Int,
        val visibleQuestCount: Int,
        val startableQuestCount: Int,
        val completedQuestCount: Int
    )
}
